### DISCUSSION TITLE
Especializações na carreira Front-End

### DISCUSSION THREAD

**Discussion by juniorconte** [frontendbr/forum#32] [Discussão]
Galera, estou passando por um tipo de "crise existencial" rsrs, e talvez vocês já tenham pensado sobre ou talvez comecem a pensar também.

Alguns profissionais se denominam Back-enders, outros Sysadmins, dai surgiram os DevOps que manjam dos paranauê de back e sys, e por fim, os Full-stacks, que além de DevOps, também sacam de front. De certa forma, isso está até bem claro, o problema, é que o Front-end em sí, se tornou algo muito amplo. Se dissermos que Front-end, na verdade é Client-side, teríamos até mobile fazendo parte da treta, mas nem vou considerar isso. Minha grande inquietação, está no fato de designers, diagramadores (HTML + CSS) e programadores JS (digo programadores que lidam com Apps com suas 4 ou 5 mil linhas de JS), acabam fazendo parte do mesmo bolo. Eu não acho isso ruim, apenas acho confuso, e muitas vezes geram aquelas discussões superficiais. Pois mesmo na roda de Front-ends, tem os caras que manjam quase nada de CSS e os caras que não tem idéia do que é MVC ou MVW*

Eu por ex, estou longe de ser um designer, conheço razoavelmente HTML e CSS, mas passo 90% de meu tempo escrevendo Apps com AngularJs.

Minha dúvida é se da pra chamar toda essa galera, de  "os caras do front", ou se existem discussões mais filosóficas que tentam dividir melhor nossos perfis.

[]'s

**Comment by mksanti** [comment]
Ta ai uma questão que não vi alguém definindo isso (se for possível dividir). 
Na visão existem 2 tipos de front, O front que puxa mais para o back e o front que puxa mais pro design. E outra... no final de contas o front-end é o cara que resolve os problemas, a chave coringa do time de desenvolvimento.

**Comment by juniorconte** [comment]
Com o advento de frameworks Js, ferramentas de scaffold, build, testes e
deploy voltadas especialmente  para o front, e ainda ferramentas como
DreamFactory, postgreREST e afins, que automatizam a construção de API's
Crudáveis, tenho visto "de perto" o Front definindo partes cada vez mais
importantes do negócio. Até começo a perceber uma espécie de inversão de
valores ou troca de lados, onde o backend começa a se tornar cada vez mais
commoditie e o front sendo o elemento que traz a diferença ou até a
dinamização do negócio.

Essa divisão entre design e programação, tem ficado cada vez mais nítida, e
no meio disso tudo, vejo os diagramadores e programadores de CSS. Ao meu
ver, frontend é um universo com 3 camadas.

**Comment by willycamargo** [comment]
Já vi as seguintes descrições:
- Front-end com conhecimento em UI e UX
- Front-end com conhecimento em AngularJS (Ou outro framework MV*)
- Front-end com conhecimento em Javascript
- Front-end com conhecimento em gráficos SVG
- Front-end com conhecimento em CMS (WP, Drupal, etc)
- Front-end etc...

Acho que é uma boa forma de definirmos o que fazemos. Trabalhamos sim no front-end do projeto, mas sempre com enfâse em áreas específicas. Dificilmente um mesmo profissional, em uma mesma empresa, irá trabalhar com front-end, back-end, banco de dados, mobile, UI, UX, SEO etc.

Sempre existirá "enfâse" em algo, até porque front-end engloba muita coisa e acredito ser impossível dominar tudo.

Aquele conceito de escrever HTML, CSS e controlar alguns poucos comportamentos com JavaScript existe apenas em algumas agências (conhecidas também como fábricas de sites), e a maioria dos profissionais têm uma passagem rápida por essas empresas. A partir do momento que começamos a trabalhar com sistemas, se torna necessário adquirirmos conhecimentos específicos (de acordo com a capacidade do profissional e necessidade da empresa).

Bem, é essa minha visão, mas gostaria de lapidar ela com a opinião de outros profissionais mais experientes.

**Comment by eduardojmatos** [comment]
Olá pessoal! Discussão bem importante essa, até porque ultimamente temos visto essa divisão voltar nas conversas e contratações das empresas.

Concordo com vocês todos. Existem frontends focados em cada um desses itens (UI, Framework, JavaScript, SVG, CMS, Layout / CSS).

Pra mim, existem dois perfis que englobam esses grupos. O **Front-end UI** e o **Front-end Engineer**.

O **Front-end UI** é focado em lidar com toda a parte visual da aplicação, e tem algumas obrigações, como:
- lidar com frameworks de layout;
- cuidar do styleguide;
- construir uma interface semântica;
- facilitar visualização de dados através de gráficos SVG/canvas;
- lidar com animações;
- participar e encabeçar das discussões de UX;
- desenvolver código acessível a pessoas com necessidades especiais;
- garantir que o progressive enhancement ou graceful degradation sejam aplicados;

Já o **Front-end Engineer** é responsável pela estrutura da aplicação. Suas responsabilidades se resumem a:
- criar ambiente de front de desenvolvimento e produção;
- definir tarefas de build;
- prover a sincronia entre back-end e front;
- escolher entre aplicar um framework de JavaScript ou não;
- criar os testes de interface/unitários;
- manter código padronizado;
- definir pattern de código;
- usar ferramentas compatíveis com browsers, lembrando sempre do cross-browser;

---

Tem mais coisas aí nessa lista, mas é um resumo de onde eu acho que cada um tem mais responsabilidades.

O que vocês acham?

**Comment by rubenmarcus** [comment]
Então o que manja de tudo é full-stack?

Pq eu posso dizer que 90% da minha experiência é com UI,mesmo eu não sendo formado em design

e 30,40% em programação voltada a back. 

Manipulação de dados (MySQL,Oracle,MSSQL,Postgre) é front tb?

Na epoca que eu comecei a mexer com sites (época de CGI E Perl e se fazia sites em newsletters), o front fazia todo o layout em iframe etable, e o back instalava os scripts e era isso. Depois veio PHP, com foruns e CMS modificados de foruns e o front (na epoca webmaster) comecou a mexer com códigos PHP Tb.

a minha dúvida é pra você ser um front requisitado nas empresas, tanto daqui quanto de fora e ganhar um salário de no mínimo 8k mensais (eu não ganho isso ,mesmo trabalhando com web por hobby desde os 12 anos (14 anos) e profissionalmente desde os 21 ( 5 anos) ), você tem que ser especialista n oque faz (UI ou JS e integrações com back ou Fullstack) ?

porque o meu perfil eu não sou especialista em UI,mas a facilidade com criação pra mim e a forma de me atualizar pra mim flui tudo mais natural. Quando mexo com ferramentas que envolvem programação, eu tb tenho facilidade só que menos. Preciso ler mais, praticar mais, tentar mais.. e acabo sendo ficando atrasado.. Eu tenho um perfil curioso e e um perfil que consegue aprender rápido e me virar sempre. Mas esse é o perfil

Hoje eu tenho 26 anos e antes dos meus 30 eu queria ter experiencias fora do País. E estou meio focado nisso.. ai não sei que só com esse perfil de criação,prototipagem, e interação com back eu conseguiria (minha lógica em JS ainda é um pouco fraca,digamos intermediaria eu preciso melhorar nisso) , ou eu preciso aprender a montar APIS, todos essas siglas ai a fundo, aprender programação a fundo e não só "pra me virar"

**Comment by rubenmarcus** [comment]
Essas que o Eduardo comentou eu tenho uma noção,mas patino ainda preciso praticar mais ( do Engineer)

-criar os testes de interface/unitários;
-criar ambiente de front de desenvolvimento e produção;
-definir tarefas de build;

Mas posso comentar pelas vagas que vejo mesmo...

vamos lá algumas

Essa da Chaordic:

https://www.linkedin.com/jobs2/view/101328962?trk=vsrp_jobs_res_name&trkInfo=VSRPsearchId%3A1467437061454859598041%2CVSRPtargetId%3A101328962%2CVSRPcmpt%3Aprimary

_Advanced knowledge in Javascript and PHP is the top most requirement;
Strong skills on Unix/Linux environment are a must;_

Essas duas eu ficaria meio assim de fazer a entrevista. Nunca mexi com Unix. e PHP ter dominio em PHP eu nao seria Front,seria des. PHP.

Monashees:
https://www.linkedin.com/jobs2/view/114574006?trk=vsrp_jobs_res_name&trkInfo=VSRPsearchId%3A1467437061454859598041%2CVSRPtargetId%3A114574006%2CVSRPcmpt%3Aprimary

_Nice To Have

Working knowledge in caching system like Varnish, Memcache etc.

Social networking API knowledge or design

Gridle, Git, Jenkins and Jira
_

Na Monashees, tudo isso que eles postaram pra mim é coisa de back. o resto ou eu tenho noção ou dominio. Mas tem coisas que eu olho e vejo, isso ai é coisa pra infra ou back. Posso aprender, já li sobre essas coisas,mas nunca coloquei em pratica.

-- na gringa --

comparando fe engineer x designer :

http://stackoverflow.com/jobs/98488/lead-software-engineer-ninja-van?searchTerm=frontend%20designer&offset=4  

Essa vaga aqui de FE Engineer só essa linha eu manjo

Frontend: HTML5/AngularJS web apps, native Android apps ( e esse native Android apps se for p programar em Java esquece)

Alias lá na gringa tudo é Engineer ou Senior Developer..
http://stackoverflow.com/jobs?searchTerm=frontend

só ver..

**Comment by eduardojmatos** [comment]
@rubenmarcus boa pergunta. Eu normalmente encaro quem diz ser full-stack como o cara que consegue jogar nos dois times, mas que tem conhecimentos mais avançados apenas em um.

Digo isso porque, se forem ver, só de competências que um front-end precisa ter já é coisa pra caramba. De back-end então, nem se fala...

Existem casos, e até conheço, de gente que consegue mandar muito bem nos dois mundos. Mas são exceções.

> Manipulação de dados (MySQL,Oracle,MSSQL,Postgre) é front tb?

Na verdade não é manipular os dados na fonte, e sim, quando eles chegam pra você, no front. Essa lógica ultimamente está sendo compartilhada com o front. Trabalho com Ember, e muita lógica de dados está presente na aplicação, em JavaScript mesmo, antes de chegar no back-end.

> a minha dúvida é pra você ser um front requisitado nas empresas, tanto daqui quanto de fora e ganhar um salário de no mínimo 8k mensais (eu não ganho isso ,mesmo trabalhando com web por hobby desde os 12 anos (14 anos) e profissionalmente desde os 21 ( 5 anos) ), você tem que ser especialista n oque faz (UI ou JS e integrações com back ou Fullstack) ?

Sobre valor de salário você precisa incluir aí o fator regional. No interior de SP, por exemplo, dificilmente você vai achar empresas pagando esse valor. Já em SP é mais tranquilo de achar.

Ser especialista no que faz já te garante algo melhor. Entendendo um pouco do mundo de back-end, e aqui vai um adendo de não precisar se fullstack, pelo menos pra conseguir conversar sobre estrutura de dados e etc., acho que facilita demais o teu objetivo.

O que tenho visto é que, as empresas tem valorizado demais perfis que não se prendem ao conhecimento de sua área. Por isso não é estranho você ver vagas requisitando PHP, Unix/Linux e etc.

Agora, há exageros mesmo.. Varnish e Memcache são coisas de back-end sim. Mas se você notar, eles colocam isso no _Nice to Have_, já sabendo que não é normal pra um front ter esses skills.

No fim das contas, as empresas e equipes avaliam seu perfil no sentido cultural. Principalmente se, caso precisem que você aprenda algo em outra área, que não seja a sua, você vai se virar bem ou vai encarar isso numa boa.

Espero não ter escrito demais hehehe Virou textão :scream: :scream:

**Comment by bernardodiasc** [comment]
Depois que client-side rendering entrou no jogo, FE tambem escreve algoritmo. Tento simplificar na minha cabeça (já que assino como frontend tb) que, BE trabalha nas coisas que tem a ver com o codebase e infra de server, linguagens de server, integrações, banco de dados, etc. FE é o cara que trabalha na apresentação e experiencia do usuario, qualquer coisa do lado cliente, seja desktop, mobile, geladeira ou gameboy.

A distinção exata não existe na pratica, mas funciona bem se o time tiver boa comunicação e souber trabalhar em equipe aproveitando as potencias de cada um. A pegadinnha aqui é que o FE que for trabalhar num projeto Rails, tem que entender da infra do rails pra poder atuar no projeto, não necessariamente atuar no BE, mas entender é importante. O mesmo vale pro BE, que pode montar toda a API e ainda assim se ver escrevendo a autenticação do lado client tendo que aprender toda uma stack como angular ou react, por exemplo.

**Comment by dionarodrigues** [comment]
Concordo com o @eduardojmatos  e com o @mikejavier :

Para mim, o front-end é separado em duas categorias:
- Front-end UI - voltados para o design, parte visual
- Front-end Engineer - voltados para o back-end

Mas penso em um front-end como um profissional flexível e capaz de se adequar aos objetivos do negócio no qual está inserido, já que, por ser uma denominação "nova" no mercado, ainda não está bem definido o que especificamente um profissional dessa área terá que fazer. 

Mas, trazendo essas denominações para nós, ou seja, para os objetivos profissionais de cada um, precisamos ter em mente qual desses caminhos queremos seguir e se aprofundar nele, não deixando de ter uma noção razoável de como as coisas funcionam do outro lado. Porém, **independente de qual caminho seguir, HTML, CSS e JS são conhecimentos fundamentais**.

**Comment by juniorconte** [comment]
Vocês diriam que design de interfaces faz parte do Front-end? 

Eu diria que sim, e imagino que existam designers que são muito bons no que fazem, só que não possuem um perfil técnico para codificar.

Se design para web faz parte do universo Front-end, não daria para colocar todas aquelas habilidades de UI/UX citadas pelo @eduardojmatos como o foco do profissional de UI/UX, pois UI/UX inclui design. Não acho que criativos de design, obrigatoriamente devam ser bons codificadores de HTML e CSS, até acho que existem bons motivos para que não sejam.

Pensando nos dois extremos, eu diria que em uma ponta temos o Engineer, que lida com regras de negócio, processos de build & deploy, segurança, etc.. e do outro lado, no extremo oposto, teríamos o designer web.

De um lado, um profissional com perfil analítico, com requisitos de ser extremamente técnico no que faz. Do outro lado, um profissional criativo, que tem o papel de encantar e permitir uma boa comunicação da interface com o humano.

No meio disso, ou quase no meio, encaixa-se o que eu chamaria de diagramador. O responsável por transcrever a interface gráfica em um código visual que seja adequado para o dispositivo / mecanismo. Cabe aqui, conhecimentos de SEO, carregamento seletivo, responsive e coisas mais. Tudo ligado a técnicas de CSS, HTML, e talvez JS (quando voltado para apresentação)

Por isso enxergo 3 camadas (esquerda, centro e direita) onde a esquerda é mais criativa, a direita é mais técnica e o centro, precisa de um pouco dos dois.

Acho importante criarmos essa distinção, para poder fazer eventos de Front-end que possam falar de design, diagramação e programação, de forma cada vez mais profunda e voltada para os vários perfis de profissionais que compreendem o Front, sem ficar frustrado por não dominar Matemática e Artes visuais ao mesmo tempo.

**Comment by elvisdetona** [comment]
Então, eu não gosto muito da sigla de "full-stack". 
E sim, existe essa "divisão do UI e do Engineer", mas se o "front" manja bem de fazer as piruetas com CSS / Semântica e também trampa com JS a fundo, integrando com o backend e cuidando de todo build? Massss, esse "cara" não é fudido no backend, saca? 
Full Stack? Apenas Front-end? Front-end developer? Cada empresa adota um nome e eu, concordo em partes com o @eduardojmatos  na separação entre UI e Engineer, mas ao mesmo tempo, não concordo muito. 
Enfim... entrei pra confundir mais o discurso.
EU me intitulo "front-end developer". e VCS?

**Comment by rickbenetti** [comment]
Também concordo com partes do @eduardojmatos e concordo muito com o @elvisdetona.

Eu por exemplo tenho usado a muito FrontEnd Designer, uma vez que meu foco não é o JS, mas tudo o que faço já deixo integrado à plataformas WordPress, Magento, CodeIgniter, C# e outros frameworks e linguagens.

Até porque conheço mais de PHP do que de JavaScript à título de exemplo.

Agora definir FrontEnd UI eu acho muito estranho, engineer também pelo que entendemos no Brasil de Engenheiro.

**Comment by felipefialho** [comment]
Bela discussão. Já tive altas conversas com o @deividmarques nesse sentido.

Sou de total acordo com o @diogorodrigues, existe sim uma certa "separação" entre os vários tipos de profissionais no mercado, porque nossa área está cada vez mais ampla e o JavaScript hoje é também uma linguagem server-side, que em tese seria responsabilidade do Back-End, mas que acaba sendo colocado como skill para o Front-End (imagina o quanto a galera do RH não se confunde?).

Mas no fim, varia de local para local. Pode ser que em determinada Startup o Desenvolvedor seja responsável por muito mais áreas enquanto em uma empresa ou agencia maior, o Desenvolvedor (agora sim) Front-End tenha a função de cuidar mais da parte de interface.

Os tipos de profissionais também são muito variados. Mas acredito que uma boa base em JavaScript (puro ou frameworks) e ter total domínio de HTML/CSS é muito necessário.

Claro que se você está dando manutenção em um projeto com tecnologia X sendo usada no Back-End (seja ela Node, .NET, Rails, PHP, não importa), você automaticamente terá que adequar ao código e a estrutura utilizada e até mesmo conseguir fazer coisas básicas nessas linguagens.

**Comment by eduardojmatos** [comment]
Boa galera, faz sentido o ponto de vocês. Eu só acho que ainda sim, existe esse perfil mais UI do que Engineer. Talvez o mercado venha "puxando" o front-end pra Engineer por conta da necessidade atual (frameworks mais avançados, interfaces mais ricas, etc.). Mas ainda assim tem aquele cara que veio do design e tem uma pegada muito mais de interface / visual do que de programação.

Talvez nas startups esse cara não tenha muita vez, porque o perfil nesses casos exige mais conhecimentos em várias áreas.

Sobre full-stack, ainda acredito no termo, e reitero que, mesmo sabendo da infra ao front, o cara sempre tem uma especialidade. Não acho que full-stack tenha que saber tudo sobre todas as coisas. Full-stack precisa entender de cada camada da aplicação, conseguir codar em todas elas e ter pelo menos uma em que ele mande muito bem. Se ele não manjar de pelo menos uma dessas áreas, aí entra naquele perfil, velho conhecido "webmaster", que fuçava em tudo, mas não tinha uma especialidade.

Bem complexo esse tema e pode variar de região pra região, empresa pra empresa...

**Comment by deividmarques** [comment]
@juniorconte  Caraca meu, vc descreveu exatamente o que estou sentindo por esses dias. Essa "crise existencial" e confesso que tem tirado meu sono. Que ótima issue essa.

Eu concordo plenamente com o que o @eduardojmatos  citou: 

_Para mim, o front-end é separado em duas categorias:_
- Front-end UI - voltados para o design, parte visual  (me vi 100% em toda descrição)
- Front-end Engineer - voltados para o back-end - (Aqui 50%)

Sempre acreditei e dialoguei muito com o @LFeh  sobre esses 2 perfis que o mercado precisa ter pra explorar ao máximo o que esses 2 perfis podem oferecer. Compreendo que nem sempre as empresas podem pagar e ter ambos os perfis nos projetos e digo infelizmente. 

Aprender outras linguagens, programar, é muito melhor e "magicamente" você é visto diferente por algumas pessoas da área.   #desabafo 

Como citei no início do texto, eu estava bem pensativo na questão do perfil, pois sempre vejo no mercado de trabalho a busca por perfil de **Engineer** e não me via em muitas descrições da vaga, e com isso me me cobrei mais a estudar, mesmo tecnologias com a qual não trabalho, porém se estudasse mais o que realmente me proponho mais a fazer com **mais** vontade acredito que seria muito mais produtivo. 

Eu sempre me pergunto até quando existirá esses 2 perfis no mercado brasileiro?

**Comment by rickbenetti** [comment]
@deividmarques acredito que tanto aqui como lá fora o FrontEnd irá se dividir em mais variantes nos próximos anos, ao menos é o que acredito.

Igual hoje existem os Backend de PHP, de Java, de C#, e tantas outras variantes.

estou errado?

**Comment by juniorconte** [comment]
A VALVE implementa um conceito bastante interessante, dizem procurar profissionais "T-shaped". Seriam os caras que tem a noção de vários assuntos "o braço do T" mas são especialistas em algum "o corpo do T". Concordo com o @eduardojmatos , e acho que Full-stack seria um profissional T-shaped. Esse tipo de profissional consegue estabelecer um diálogo com várias partes do projeto, e também agregar valor no que é um especialista, sendo assim, faz sentido que busquemos ao máximo possível ser Full-stacks.

Quanto as divisões, depois de ler tudo de novo e refletir, também acho mais sensato pensar que existem dois perfis principais, um proveniente da linhagem Design e outro da Programação. Talvez nos caiba o conceito de Arquiteto e Engenheiro que é empregado na construção civil http://goo.gl/c9lfkh

Por outro lado, citando o que o @rkb81 disse, acho que inevitavelmente, com o passar do tempo teremos mais divisões, assim como ocorre na medicina. Pois temos assuntos latentes, que por sí só são bem complexos, e merecem atenção: Real Time, Off-line data, responsive, etc. Acho que dessas áreas sairiam alguns especialistas.

Até que ponto vocês acham que seria válido pensar em eventos de Front-end separados em duas trilhas, uma voltada para o "UI" e outra para "Engineer" ? Digo isso, visando criar discussões mais profundas dentro de cada grupo/trilha.

**Comment by dionarodrigues** [comment]
No fim das contas, o front-end é um camaleão. Hehehe

**Comment by dionarodrigues** [comment]
@juniorconte , pensando na questão da divisão entre os 02 perfis de front-end, talvez pode ser interessante separar os eventos.

No entanto, penso que o contraste é o causador do equilíbrio nos eventos. É muito interessante ter uma visão do todo na área, assistir palestras que abrangem mais engineer do que UI e vice-versa. Afinal, como front-end, precisamos estar preparados para lidar com o máximo de técnicas possíveis (isso não quer dizer se especializar), ainda mais em um mercado que ainda está amadurecendo nesse quesito.

**Comment by elvisdetona** [comment]
Falando em divisão de trilhas em eventos... sei lá, eu não gosto muito dessa idéia. Quando tem divisão, SEMPRE alguem perde algo. Quando EU vou a eventos, eu quero absorver o máximo de conhecimento. Um exemplo básico seria o evento "brazil JS". Só "engineer js" que deve ir ao evento? Um cara de UI não iria? Enfim... o evento é para DESENVOLVEDORES, muito mais front-end do que backend. Mas sim, tem coisas que são pro perfil de backend... mas aí os front tb não poderiam ir? Hehehe... enfim, tem gente que gosta de trilhas, mas acho que dividir o front-end é meio arriscado.
Nós pertencemos a uma comunidade cheia de divisoes e grupinhos e isso gera "briguinhas" desnecessarias... imagina se os eventos de front dividem em trilhas? 
Sala "front junior', front pleno, front css, front ux, front semantica, front acessibilidade, front node, front angular vs react... 
#medo disso acontecer um dia!

**Comment by juniorconte** [comment]
Eu não acho que trilhas criariam um impeditivo para um engineer participar de UI ou vice-versa. Desde que o cara queira, bora lá

Acho que trilhas permitiriam o espectador ter uma noção mais clara do que ele irá encontrar no espaço X ou Y.

Isso não cria divisões, mas sim segmentações. Segmentar é importante para poder criar discussões mais profundas, que não fiquem sempre no "como iniciar com XPTO". Embora talks de "como iniciar" sejam importantes, sinto que existe uma pressão antes de qualquer talk, introduzir um conteúdo que para metade dos caras ali já é maçante pra caramba, e pra outra metade, é um bicho de 7 cabeças. Sinto muita dificuldade em preparar uma talk de 40 minutos sobre AngularJs ou Automação por ex, sabendo que na platéia tem desde designers até back-enders que de alguma forma caíram no front. Parece que nunca dá pra aprofundar o diálogo, os exemplos, etc. Sinto que isso tem sido um tipo de "freio" para surgirem especialistas.

Especialistas são muito, muito importantes. São esses caras que geralmente são capazes de sacar e desenvolver conceitos / ferramentas mais inovadores, pois conhecem profundamente X ou Y. Ao mesmo tempo, que "Full-stacks" são importantes, para manter o dialogo num nível mais alto em todas as etapas do projeto.

Acho que trilhas não são válidas para todos os eventos, na verdade a maioria são pequenos e não caberiam trilhas... Mas talvez, fazer eventos com algumas edições focadas. Por ex, se rola um Femug na sua cidade, talvez fazer um encontro mais voltado para o UI, outro mais para o Engineer, e por ai vai.

**Comment by elvisdetona** [comment]
@juniorconte  Evento com trilha não curto muito. Mas pra focar, existem os meetups... um assunto e aprofunda... enfim, acho valido tb tem eventos com trilhas. EU pessoalmente não gosto.

**Comment by juniorconte** [comment]
@elvisdetona no fundo eu tbm sinto que estou perdendo algo por não poder estar em duas trilhas ao mesmo tempo :(

Talvez meetups temáticos seja uma saída menos dolorosa.

**Comment by rickbenetti** [comment]
@elvisdetona acredito que o problema das trilhas é focar os eventos todos no mesmo dia, eu gostava antigamente dos eventos de 2 dias por conta disso, um dia era focado numa temática e outro dia em outra temática, assim todos podia assistir a todas se desejassem, sem perda e como muito mais relevância podendo complementar apresentações devido a temática.

Mas acredito que a organização dos eventos deveria ser discutida se assim for melhor em outro post específico sobre o tema.

**Comment by felipefialho** [comment]
Eu odeio fazer escolhas, portanto odeio trilhas :p

**Comment by felipecodes** [comment]
Se todas essas definições confundem até profissionais mais experientes, imagina as pessoas que estão começando. Eu fico mais perdido que cego em tiroteio! o/

Como disse o @eduardojmatos, existe os dois tipos: O **Front-end UI** e o **Front-end Engineer**. Vocês saberiam dizer onde cada um desses profissionais é mais requisitado(startups, agências etc...)?

**Comment by felipefialho** [comment]
@felipoliveira Depende muito do perfil da empresa. O ideal é ter os dois tipos de profissionais, mas nem todas tem o R$ necessário para isso.

Então acredito que um Front-End Engineer tem mais chances no mercado de trabalho, visto que pode realizar ambas as funções (mesmo que seu foco não seja necessariamente a interface).

**Comment by rickbenetti** [comment]
Hoje o Brad Frost  lançou um texto que explica justamente o que faço e venho falando há tempos: [FrontEnd Design](http://bradfrost.com/blog/post/frontend-design/)

**Comment by felipefialho** [comment]
Muito bom @rkb81. Acho que me enquadro próximo disso também.

**Comment by juniorconte** [comment]
Acho que o Brad Frost conseguiu sintetizar muito bem o 3º perfil. Eu havia chamado de diagramador, mas "Frontend design" me parece bem adequado. @rkb81 Vlw por compartilhar

**Comment by dionarodrigues** [comment]
+1 @rkb81 // Sou Front-end Designer

**Comment by elvisdetona** [comment]
@diogorodrigues  mas vc é DESIGNER de verdade mano! Não vale! ahahaha, eu por exemplo... sou formado em design, mas se o job passar de uma casinha com uma árvore cheia de maçãs, eu to ferrado. Se formar em design foi bom, pq eu tenho noções de cores, perspectiva... :art:

**Comment by felipefialho** [comment]
HAHAHAHAHAHAHA, é igual eu @elvisdetona, consigo muito bem dar continuidade a um design direto no código, mas se pedirem para criar algo do 0, FUDEU.

O @diogorodrigues manda bem demais como designer!

**Comment by tcelestino** [comment]
Copy-Paste Engineer. :smile: 

Estou mais para UX, mas venho algum tempo estudando bastante sobre performance (front end com foco em performance, existe?)

**Comment by deividmarques** [comment]
@tcelestino penso assim: quem trabalha com front precisa ter performance no core, independente do perfil que ele seja.

**Comment by dionarodrigues** [comment]
Obrigado @elvisdetona e @LFeh Hehehehe

**Comment by eduardojmatos** [comment]
Olha só hein? Parece que não é só a gente que estava pensando nessa divisão de "tipos" de front =D

**Comment by fabriciofmsilva** [comment]
Vou deixar aqui minha opinião. Acredito que a galera tenha resolvido bem a discussão. No fim concordo muito com o que a galera disse, especialmente o @eduardojmatos que separou o front em UI e Engineer.

Quando penso em front-end, penso em HTML, CSS e JavaScript, acho essencial todo Front-End Developer saber trabalhar com no mínimo essas três tecnologias. É claro que assim como em outras áreas vamos nos especializando naquilo que mais temos afinidade. Ou até combinando conhecimentos, como por exemplo um cara que entender de Design acaba focando mais em criar interfaces ricas, com boa UX e boa acessibilidade, animações incríveis, etc. Enquanto o cara que manja mais de programação acaba partindo mais pro JS, ficando responsável por fazer as devidas conexões com o Back-End, cache off-line de dados, etc.

Penso que devemos ser ambos, um Front-End generalista, que tem a visão do todo, inclusive do Back-End (preferencialmente na tecnologia que sua empresa trabalha). E a partir daí nos especializarmos naquilo que temos mais afinidade. Com isso ganhamos uma visão do todo, o que faz sermos um especialista melhor. No fim somos todos Front-End Developers, como disse o @elvisdetona !

**Comment by tcelestino** [comment]
@deividmarques, mas performance é algo que bem profundo. Você precisa ter noção, mas fazer pesquisa, testes, implementações etc... pode ser algo bem técnico também. Não sei se expliquei corretamente. :smile:

**Comment by rubenmarcus** [comment]
Stack Overflow Developer ahahha

**Comment by tcelestino** [comment]
@eduardojmatos, salvei sua definição de "front-end engineer", para uma consulta posterior.

**Comment by diessica** [comment]
"Front-end UI" é um termo bem estranho, pois ele não tem um agente ("Developer", "Engineer", "Designer").

**Comment by diessica** [comment]
Fiquei bastante inspirada com essa discussão e resolvi escrever sobre o assunto, em detalhes, com base nas minhas visões:
https://diessi.ca/blog/a-fantastica-diversidade-do-front-end/

**Comment by eduardojmatos** [comment]
@diessica mandou bem demais! Eu entendi seu ponto sobre ser estranho o termo Front-end UI. No fundo, o que o Brad Frost nomeou faz um pouco mais de sentido. Gostei tbm da sua sugestão de "Front-end UI Developer".

Aplaudi de pé aqui quando você cita que, normalmente, as empresas não sabem que tipo de profissional elas querem. Isso causa frustração, tanto para o profissional como pra empresa, pois ambos esperavam algo diferente.

Enfim, belo post, daqueles pra passar pra frente quando alguém perguntar sobre o assunto <3 <3 <3

**Comment by fdaciuk** [comment]
Muito bom mesmo @diessica! Esse post merece ir para o README do repositório de vagas, como referência :clap: :clap:

**Comment by diessica** [comment]
@eduardojmatos @fdaciuk Obrigada, pessoal!

Estava há tempos para escrever um artigo reunindo meus pensamentos sobre isso. É o meu ponto de vista, então fico bastante feliz que esteja compatível com o de vocês!

No fim, a gente tenta padronizar as coisas, rotular e tal, mas isso é fantasioso. Desde que a gente se agarre aos nossos interesses, tem espaço sim no mercado. :)

**Comment by felipefialho** [comment]
@diessica Fantastico seu artigo. :clap: :clap: :clap: 

Não tinha como concordar mais, quando me perguntarem minha opinião sobre o assunto vou encaminhar o seu texto.

Em tempo, te coloquei nessa feed-list que mantenho -> https://github.com/LFeh/feed-list/blob/master/README.md

**Comment by lagden** [comment]
@diessica segundo seu artigo eu sou um :godmode: 
uhahuahu... **sqn** :finnadie:

**Comment by Marcelosilva10** [comment]
Entrando nessa pauta, de ser Front end UI e Font end enginner, podemos separar alguma linguagem de programação separadas em seguintes áreas como exemplo, Front end ui - Javascript e Enginner com Php server side?
Pergunto porque tem muitas agências que você pode saber JS, mais o primordial para eles e o PHP principalmente na região de Sorocaba,

**Comment by eduardojmatos** [comment]
@Marcelosilva10 eu acho que não cara. Uma coisa é a conversa sobre ser FullStack (discussão #135) e outra é dominar conhecimentos sobre front-end. O Frontend UI (citado pela @diessica) é mais voltado a áreas de conhecimento de interface, experiência de usuário e design. O Engineer em conhecimentos de programação, estrutura da aplicação e algoritmos.

**Comment by paulo36** [comment]
qual a diferença do engenheiro e do desenvolvedor frontend?O que é bom estudar?Pois estudo tais linguagens,etc e vou procurar vagas vejo outras que não conhecia.

**Comment by andrecoliveira** [comment]
Oi @paulo36, dê uma lida nesse artigo [AQUI](https://medium.com/tableless/qual-perfil-de-um-front-end-engineer-7833509ebd40).

**Comment by cristofersousa** [comment]
> Ressuscitando esse tópico

Estou procurando sobre uma Pós focada mais em Desenvolvimento Front-end e encontrei essa da IGT 
> https://www.igti.com.br/cursos/mba-em-desenvolvimento-front-end/  

Alguém conhece ou tem informações sobre essa Pós, parece ser bem introdutória mas já é um grito no escuro ao menos. Sabendo que se trata de um MBA o foco é 100% mercado então espero encontrar ali pessoas com experiências práticas, durante minha graduação percebi que somente professor que tinha vivido no mercado conseguiam conduzir a disciplina para um modelo mais real e não ficar na alcunhas dos livros.

Acredito que para uma pós, ainda mais especifica para em front-end, adoraria ver mais padrões, arquitetura, modelos de implementação além de abordar mais sobre reatividade, testes de todos os tipos possíveis, escalabilidade e performance além de acessibilidade.  Abordar de es5 (legado) para cima além de alguns boilerplates. Tanto o livro do Eric e Martin Fowler parece cobrir bastante isso:

> https://leanpub.com/composingsoftware
> https://martinfowler.com/articles/refactoring-2nd-ed.html

Acho que um curso com essa visão é difícil pois os professores dessa área na area acadêmica ainda não existem, talvez seria um apanhado de vários profissionais que temos aqui no forum que se juntassem para montar esse curso e disseminar essas disciplinas seria algo bem impactante para o mercado. 

É impossível não entrar nos frameworks pois abordamos o que o mercado está utilizando então acredito que ainda assim teria que ter uma disciplina focada para isso criando um webapp com Angular, React, Vue e Ember. Um estudo minucioso sobre o que esses caras cobrem e suas principais diferenças afinal se existe um fw por mês todos querem cobrir uma dor ou um problema o que muda o conceito é a adesão da comunidade.

Ainda acho que seria válido adotar uma disciplina para tratar de cenários caóticos, bugs, código legado enfim vamos fazer engenharia reversa e melhorar a qualidade de vida do software. 

> https://leanpub.com/legacycode






Enfim é isso, a busca pelo conhecimento não cessa nunca.

**Comment by tarciozemel** [comment]
Ressuscitando de novo: https://desenvolvimentoparaweb.com/miscelanea/a-grande-divisao-front-end/
